gem/refactorise.py

```diff
@@ -112,6 +116,135 @@ def product(*args):
             result.add(sum_indices, atomics, rest)
         return result
 
+    def argument_indices_extent(self, factor):
+        if self.argument_indices is None:
+            raise AssertionError("argument_indices property not initialised.")
+        return numpy.product([i.extent for i in set(factor.free_indices).intersection(self.argument_indices)])
+
+    def all_sum_indices(self):
+        result = []
+        collected = set()
+        for (sum_indices_set, _), (sum_indices, _) in iteritems(self.ordering):
+            if sum_indices_set not in collected:
+                result.append((sum_indices_set, sum_indices))
+                collected.add(sum_indices_set)
+        return result
+
+    def to_expression(self):
+        # No argument factorisation here yet
```

Comment:
This does more than just converting the monomial to gem, right?  Also, docstring.